pd/src/state.rs

```diff
@@ -112,6 +109,34 @@ ON CONFLICT (id) DO UPDATE SET data = $1
             .await?;
         }
 
+        if block.is_epoch_boundary.is_none() {
+            return Err(anyhow::anyhow!("EndBlock must be called prior to Commit, `is_epoch_boundary` was not set on the pending block"));
+        }
+        if block.is_epoch_boundary.unwrap() {
+            // validator rates need updating on epoch boundaries
+            let validators = self.validators().await?;
+            for validator in validators {
+                tracing::info!("updating validator rates for validator: {:?}", validator.0);
+                // TODO @ava insert calls here
+                let validator_rate = 0;
+                let voting_power = 0;
+
+                let pubkey_str = serde_json::to_string(&validator.0)?;
```

Comment:
non-blocking on this PR: should we try to use bech32 for validator identities consistently? (we can't use it in the DB at the moment because there's no round trip parsing upstream...)